### PR TITLE
MVP phase 1 - Add database

### PR DESCRIPTION
**Pull request by josecabrera13**
What is in the PR ?

Create database
Create database seed worker to populate the database
Create filter for countries

### CODE REVIEW

File: app/src/main/res/layout/fragment_country_hub.xml [hunk 1]
```diff
@@ -5,6 +5,15 @@
...
 
+    <androidx.appcompat.widget.SearchView
+        android:id="@+id/search_view"
+        android:layout_width="0dp"
+        android:layout_height="50dp"
+        app:iconifiedByDefault="false"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
```
Comment (MauricioNavaC) [inline app/src/main/res/layout/fragment_country_hub.xml:16]: Will be good to use dimens resources in **width** and **height** properties, I added the 0 in my PR but the 50 one has to be added.
_Maybe you'll get a conflict error here after pull my changes._

File: app/src/main/java/com/globant/countriesappglb/data/CountryDao.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import androidx.room.Query
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface CountryDao {
+    @Query("SELECT * FROM countryentity ORDER BY commonName")
+    fun getAllCountries(): Flow<List<CountryEntity>>
+
+    @Query("SELECT * FROM countryentity WHERE commonName LIKE :countryName")
+    fun getCountriesFilteredByName(countryName : String) : Flow<List<CountryEntity>>
```
Comment (MauricioNavaC) [inline app/src/main/java/com/globant/countriesappglb/data/CountryDao.kt:16]: I don't know if will be good to filter using both names "common and official" in this query.

File: app/src/main/res/layout/fragment_country_hub.xml (hunk 1 above)
Comment (josecabrera13) [reply app/src/main/res/layout/fragment_country_hub.xml]: The conflict is resolved 👍

File: app/src/main/java/com/globant/countriesappglb/data/CountryDao.kt (hunk 2 above)
Comment (josecabrera13) [reply app/src/main/java/com/globant/countriesappglb/data/CountryDao.kt:16]: sure let's ask to iOS team how are they handling the search.

Comment (MauricioNavaC) [reply app/src/main/java/com/globant/countriesappglb/data/CountryDao.kt:16]: Okay!

**Review by MauricioNavaC — APPROVED** [review]

**Pull request merged**